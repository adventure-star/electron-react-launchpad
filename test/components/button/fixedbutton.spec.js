import React from 'react';
import ReactTestRenderer from 'react-test-renderer';
import FixedButton from '../../../src/renderer/components/button/fixedbutton'

function testFunction() {
    // DO NOTHING

    return "";
}

function setupStandard() {
    const renderer = ReactTestRenderer.create(<FixedButton onClick={testFunction} className='testClass'>Test button text</FixedButton>);
    const component = renderer.toJSON();

    return {
        component
    };
}

describe('Button - FixedButton component', () => {
    it('should be of type div', () => {
        const { component } = setupStandard();
        expect(component.type).toBe('div');
    });

    it('should have only the classes "btn-fixed', () => {
        const { component } = setupStandard();
        expect(component.props.className).toBe('fixed-action-btn');
    });

    it('first child should have only the classes "btn-floating" and "testClass"', () => {
        const { component } = setupStandard();
        expect(component.children[0].props.className).toBe('btn-floating testClass');
    });

    it('first child should have an onClick function', () => {
        const { component } = setupStandard();
        expect(component.children[0].props.onClick).toBe(testFunction);
    });

    it('children should be the text "Test content"', () => {
        const { component } = setupStandard();
        expect(component.children[0].children[0]).toBe('Test button text');
    });

    it('Matches/Renders a snapshot', () => {
        const { component } = setupStandard();
        expect(component).toMatchSnapshot();
    });
});
