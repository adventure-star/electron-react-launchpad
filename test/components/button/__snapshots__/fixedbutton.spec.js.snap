// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button - FixedButton component Matches/Renders a snapshot 1`] = `
<div
  className="fixed-action-btn"
>
  <button
    className="btn-floating testClass"
    onClick={[Function]}
  >
    Test button text
  </button>
</div>
`;
